src/Queue/RabbitMQQueue.php

```diff
@@ -204,6 +201,10 @@ public function pop($queue = null)
         try {
             $queue = $this->getQueue($queue);
 
+            $this->declareExchange($queue);
+            $this->declareQueue($queue, true, false);
+            $this->bindQueue($queue, $queue, $queue);
+
```

Comment:
Same as my previous comment.

A Consumer `$this->connection->pop('queue')` should not create exchage, queue or bindings.

I know there is an issue:
When a queue does not exits, the active channel is closed. The consumer is looping and can't get messages.
Resulting in an worker/consumer loop. (Death worker)

To solve this issue see: [Link](https://github.com/adm-bome/laravel-queue-rabbitmq/blob/8a21c99bb7ba54da3a48f7196e3f27279a7eae79/src/Queue/RabbitMQQueue.php#L256)

Reply:
maybe you can up vote my pull request #304 to solve the issues